boot/seal.go

```diff
@@ -212,24 +285,87 @@ func resealKeyToModeenv(rootdir string, model *asserts.Model, modeenv *Modeenv,
 		bootloader.RoleRunMode:  bl.Name(),
 	}
 
+	authKeyFile := filepath.Join(dirs.SnapFDEDirUnder(rootdir), "tpm-policy-auth-key")
+	if err := resealRunObjectKeys(pbc, authKeyFile, roleToBlName); err != nil {
+		return err
+	}
+	logger.Debugf("resealing (%d) succeeded", nextCount)
+
+	// reseal the fallback object
+	rpbc := toPredictableBootChains(recoveryBootChains)
+
+	var nextFallbackCount int
+	needed, nextFallbackCount, err = isResealNeeded(rpbc, recoveryBootChainsFileUnder(rootdir), expectReseal)
+	if err != nil {
+		return err
+	}
+	if needed {
+		rpbcJSON, _ := json.Marshal(rpbc)
+		logger.Debugf("resealing (%d) to recovery boot chains: %s", nextCount, rpbcJSON)
+
+		if err := resealFallbackObjectKeys(rpbc, authKeyFile, roleToBlName); err != nil {
+			return err
+		}
+		logger.Debugf("fallback resealing (%d) succeeded", nextFallbackCount)
+	}
+
+	bootChainsPath := bootChainsFileUnder(rootdir)
+	if err := writeBootChains(pbc, bootChainsPath, nextCount); err != nil {
+		return err
+	}
+
+	recoveryBootChainsPath := recoveryBootChainsFileUnder(rootdir)
+	if err := writeBootChains(rpbc, recoveryBootChainsPath, nextFallbackCount); err != nil {
```

Comment:
this write should go under the `if needed ...{`, no need to write the file if we didn't detect that reseal was needed, we do the same for the main one